static/js/browser2/statvar_hierarchy.tsx

```diff
@@ -130,39 +166,75 @@ export class StatVarHierarchy extends React.Component<
       })
       .then((resp) => {
         const data = resp.data;
-        const statVarGroups = data["statVarGroups"];
-        const statVars = data["statVars"];
+        const svgInfo = data["statVarGroups"];
+        const svInfo = data["statVars"];
         removeSpinner(LOADING_CONTAINER_ID);
         this.setState({
-          statVarGroups,
-          statVars,
+          svgInfo,
+          svInfo,
         });
+        const svPath = {};
+        if (this.props.svs) {
+          for (const sv of this.props.svs) {
+            svPath[sv] = this.getPath(sv);
```

Comment:
It looks like the getPath function uses state.svgInfo and state.svInfo and those two states are set in this same function. However, setState is an asynchronous function that may not update the state right away, so this could be problematic. Maybe update the getPath function to take in the actual svgInfo and svInfo.

Reply:
Good catch. As svInfo and svgInfo is constant (only set once in componentDidMount), I moved them out of state to simplify the logic.